gateway/src/apicast/policy/3scale_batcher/keys_helper.lua

```diff
@@ -0,0 +1,79 @@
+local ipairs = ipairs
+local format = string.format
+local insert = table.insert
+local concat = table.concat
+local sort = table.sort
+local unpack = table.unpack
+local ngx_re = ngx.re
+local table = table
+
+local _M = {}
+
+local function creds_part_in_key(creds)
+  if creds.app_id and creds.app_key then
+    return format("app_id:%s,app_key:%s", creds.app_id, creds.app_key)
+  elseif creds.user_key then
+    return format("user_key:%s", creds.user_key)
+  elseif creds.access_token then
+    return format("access_token:%s", creds.access_token)
+  end
+end
+
+local function metrics_part_in_key(usage)
+  local usages = table.new(#usage.metrics, 0)
```

Comment:
`table.new` by itself should a `nil`. Is this tested?

`table.new` has to be required, like https://github.com/openresty/lua-resty-core/blob/c258785f2173ae89e2fe1d34078e5eea375d4a2b/lib/resty/core/base.lua#L46-L49

or you can use the one from the `resty.core.base` like `local new_tab = require('resty.core.base').new_tab`.

exposed from there; https://github.com/openresty/lua-resty-core/blob/c258785f2173ae89e2fe1d34078e5eea375d4a2b/lib/resty/core/base.lua#L137

Reply:
It's tested and it works in my dev environment, but now I'm not sure why.

Reply:
`require('table.new')` will make `table.new` available
and that is required by `resty.core` in `init` phase.

So all good 👍